lib/matplotlib/colorbar.py

```diff
@@ -579,6 +579,10 @@ def _reset_locator_formatter_scale(self):
             # mid point is easier.
             self.ax.set_xscale('log')
             self.ax.set_yscale('log')
+
+            if type(self.norm) == colors.LogNorm:
```

Comment:
OK, actually we don't need the check here, do we?

Reply:
That is a very good point.